src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -1013,6 +1013,7 @@ protected void postRouteProcess() {
             SitePinInst altSource = net.getAlternateSource();
             SiteInst si = source.getSiteInst();
             boolean altSourcePreviouslyRouted = altSource != null ? altSource.isRouted() : false;
+            boolean sourcePreviouslyRouted = source != null ? source.isRouted() : false;
```

Comment:
It's an unwritten invariant that `source != null` -- perhaps we should have an `assert()` here just in case.
(It comes up later too)